indico/modules/events/contributions/controllers.py

```diff
@@ -375,3 +378,69 @@ class RHContributionsExportPDFBookSorted(RHManageContributionsActionsBase):
     def _process(self):
         pdf = ContributionBook(self._conf, session.user, self.contribs, tz=self.event_new.timezone, sort_by='boardNo')
         return send_file('book_of_abstracts.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHManageContributionTypes(RHManageContributionsBase):
+    """Dialog to manage the ContributionTypes of an event"""
+
+    def _process(self):
+        contrib_types = self.event_new.contribution_types.all()
+        return jsonify_template('events/contributions/management/types_dialog.html', event=self.event_new,
+                                contrib_types=contrib_types)
+
+
+class RHManageContributionTypeBase(RHManageContributionsBase):
+    """Manage a contribution type of an event"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib_type
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageContributionsBase._checkParams(self, params)
+        self.contrib_type = ContributionType.get_one(request.view_args['contrib_type_id'])
+
+
+class RHManageContributionType(RHManageContributionTypeBase):
+    """Dialog to edit a ContributionType"""
+
+    def _process(self):
+        form = ContributionTypeForm(self.event_new, name=self.contrib_type.name,
+                                    description=self.contrib_type.description)
+        if form.validate_on_submit():
+            self.contrib_type.populate_from_dict(form.data)
+            self.event_new.log(EventLogRealm.management, EventLogKind.change, 'Contributions',
+                               'Updated type: {}'.format(self.contrib_type.name), session.user)
```

Comment:
save the same before populating? logging the new name here in case of a rename is a bit weird